graphql_compiler/schema_transformation/rename_schema.py

```diff
@@ -92,6 +91,13 @@
 - If you suppress a type Foo, no other type Bar may keep fields of type Foo (those fields must be
   suppressed). However, if type Foo has a field of that type Foo, it is legal to suppress type Foo
   without explicitly suppressing that particular field.
+- If you suppress a type Foo implementing an interface Bar, then Bar will remain in the schema but
+  not in the root query type, making it unqueryable. This is to prevent situations where a scope in
+  a query is of type Bar without also being of some more specific type, which may yield vertices of
+  type Foo even though it was suppressed. For this reason, all interfaces that Bar implements will
+  also be made unqueryable in the same way, and this removal from the root query type will happen
+  recursively until all interfaces that Foo implements (either directly or indirectly) are
+  unqueryable.
```

Comment:
Hm, so this is a bit tricky-- the plan for field renamings in PR 1005 is more like allowing something that should already exist, specifically if we have a schema like

```
# Exists interface named Foo with a single field named id
type Bar implements Foo {
  name: String
  id: String
}
```

then it should be legal to rename the `name` field, just like if `Bar` implemented no interfaces. In other words, it seems like it's less of a large functionality change that's coming soon and more like allowing something that should be allowed without adding more special cases to the field renaming API, had the `NotImplementedError` checks not been so strict.

Reply:
Is the case in the test case I linked above going to be implemented eventually or will it always be a `CascadingSuppressionError`? If it is going to be implemented eventually, maybe in the `Operations that are not yet supported but will be implemented:` section there should be something like `automatic suppression of fields when the field is of a suppressed interface type`?

Reply:
The current implementation will always be a `CascadingSuppressionError`-- the problem in that test case is that the `Giraffe` type's `friend` field is of type `Character`, and `Character` is to be made unqueryable. However, there's no way to suppress any fields in `Giraffe` right now because `Giraffe` implements an interface, so until the changes in #1005 relax this restriction, there's no way to implement the part of the test that shows how we'd want to do such a thing.

Reply:
Right, I'm not proposing you implement in this PR, but I think it should be documented somewhere that you can't suppress an object that implements  an interface if that interfaces is a field type on another object. Is that already documented somewhere that I missed?